### ISSUE TITLE
Thank you for attention.

### ISSUE THREAD

**Issue by guangdongliang** [huangyangyu/NoiseFace#5]
Thank you for attention.

Our method is an angular-based method. In your pipeline, please make sure the output of fullconnected-layer is the cosine value(In other words, the input and weights of fc layer both are normalized). You can use the debugging information to debug.

In our training process, we firstly trained the L2-softmax model called M, then trained the arcface model by using corresponding M as pretrained model.

L2-softmax structure can be referred to: [L2-softmax deploy file](https://github.com/huangyangyu/NoiseFace/blob/master/deploy/resnet20_l2softmax_train.prototxt)
Arcface structure can be referred to: [arcface deploy file](https://github.com/huangyangyu/NoiseFace/blob/master/deploy/resnet20_arcface_train.prototxt)

I suggest you use our r20 to do experiment, then replace the r20 with r100.

Hope you can get success.
If you reproduce the result, please let us known.

_Originally posted by @huangyangyu in https://github.com/huangyangyu/NoiseFace/issues/4#issuecomment-487490440_
